joomla/joomla-cms
# [4.0] Searchtools problems Fix

Pull Request for Issue #21676.

### Summary of Changes
The "default value" for list limit moved out of the list-container. The default value  comes from the Javascript and not from the filter-form. I'm not quite happy with this solution because it makes it impossible to exclude the list values from being cleared. (sorting and number of items). To achieve that a change of HTML-code and CSS-styling would be neccessary . I know now why CSS-styling shouldn't use the Javascript selectors.


### Testing Instructions
Before installing this fix:
- Go to the Plugin List or any other list with more than ca 30 items. 
- Check the list limit, probably 20 or 25.
- Click on the Clear button.
- All items will appear in the list and with a wrong list limit.
After patch install and JS rebuild.
- Do the same.
- The number of items is according to the filter form setting.
- The list limit is set to the filter form setting
Difference check
- Edit the default value for list limit in the filter form
e.g. in administrator/components/com_plugins/forms/filter_plugins.xml
Change line 78 to default="5"
-> If you use a non standard value e.g. edit to
append="33"
default="33"

Rerun the list and click clear.
The list limit should now be 5. (be aware - list could have bin cached)
Extra check - set the value in the view-tmpl
add: $searchtoolsOptions = array('defaultLimit' => 50);
edit to : 
echo LayoutHelper::render('joomla.searchtools.default', array('view' => $this,'options' => $searchtoolsOptions));
Rerun and the value after Clear should be 50.

### Actual result
Shows all items.


### Documentation Changes Required
No
